lib/elixir/lib/kernel.ex

```diff
@@ -2805,11 +2805,16 @@ defmodule Kernel do
   ## Module names
 
   A module name can be any atom, but Elixir provides a special syntax which is
-  usually used for module names. What is called a module name is an upper case
-  character followed by any alphanumeric characters or `_` or `.`. This
-  identifier is equivilant to an atom prefixed by `Elixir.`. So in the
+  usually used for module names. What is called a module name is an _upper case
+  Latin letter_ followed by any number of _Latin letters_, _numbers_, _underscores_
+  or _periods_.
+  This identifier is equivilant to an atom prefixed by `Elixir.`. So in the
   `defmodule Foo` example `Foo` is equivalent to `:"Elixir.Foo"`
 
+  A regular expression representing this syntax would be:
+
+      ~r/^[A-Z][a-zA-Z0-9_.]*$/
```

Comment:
This regular expression isn't correct:
```
Regex.match?(~r/^[A-Z][a-zA-Z0-9_.]*$/, "Q..Q")
#=> true
Range.range?(Q..Q)
#=> true
```
Do we really need to provide it at all?